course/models.py

```diff
@@ -722,17 +723,28 @@ def clean(self):
             errors['opening_time'] = _("Opening time must be earlier than the closing time.")
         if self.late_submissions_allowed and self.late_submission_deadline <= self.closing_time:
             errors['late_submission_deadline'] = _("Late submission deadline must be later than the closing time.")
+        if self.reading_opening_time and self.reading_opening_time > self.opening_time:
+            errors['reading_opening_time'] = _("Opening time of reading material "
+                "must be earlier than the opening time of the exercises.")
         if errors:
             raise ValidationError(errors)
 
     def is_open(self, when=None):
         when = when or timezone.now()
+        if self.reading_opening_time:
+            return self.reading_opening_time <= when <= self.closing_time
         return self.opening_time <= when <= self.closing_time
 
     def is_after_open(self, when=None):
         """
         Checks if current time is past the round opening time.
         """
+        when = when or timezone.now()
+        if self.reading_opening_time:
+            return self.reading_opening_time <= when
+        return self.opening_time <= when
+
+    def exercises_are_after_open(self, when=None):
```

Comment:
`exercises_are_after_open` is a bad name for the method. How about `have_exercises_been_opened`?